(DE 1355) Migration old analytics to dbt models

lricard-pass opened the pull request:
**Comment** : @lrainteau-pass j'ai ajouté deux lignes dans le Makefile : 
1. pour linker son compte google du pass et gcloud (`gcloud auth application-default login`)
2. mettre jour les components gcloud (`gcloud components update -q`)

**Comment** : @Jules-Arbelot j'ai ajouté modèles dbt, chaud de tes retours
1. `int_firebase__native_booking`
2. `int_firebase__native_visit`

[Ticket Notion](https://www.notion.so/passcultureapp/Migration-mod-les-dbt-int-mart-1b3ad4e0ff9880969285c8e45679ebc1?pvs=4)

lricard-pass committed:
feat(dbt): Enhance analytics installation process

- Add gcloud authentication step
- Update gcloud components
- Include additional setup instructions for DBT environment
- Migrate dbt model int_firebase__native_booking

lricard-pass committed:
feat(dbt) Migrate firebase_visit to int dbt model

lricard-pass committed:
Migrate old analytics models to dbt int

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            any_value(traffic_gen) as traffic_gen,
+            any_value(traffic_content) as traffic_content,
+            max(event_timestamp) as last_event_timestamp,
+            countif(event_name = 'ConsultOffer') as nb_consult_offer,
```
```suggestion
            countif(event_name = 'ConsultOffer') as total_consulted_offer,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            any_value(traffic_content) as traffic_content,
+            max(event_timestamp) as last_event_timestamp,
+            countif(event_name = 'ConsultOffer') as nb_consult_offer,
+            countif(event_name = 'BookingConfirmation') as nb_booking_confirmation,
```
```suggestion
            countif(event_name = 'BookingConfirmation') as total_confirmed_booking,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            max(event_timestamp) as last_event_timestamp,
+            countif(event_name = 'ConsultOffer') as nb_consult_offer,
+            countif(event_name = 'BookingConfirmation') as nb_booking_confirmation,
+            countif(event_name = 'HasAddedOfferToFavorites') as nb_add_to_favorites,
```
```suggestion
            countif(event_name = 'HasAddedOfferToFavorites') as total_added_to_favorites,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'ConsultOffer') as nb_consult_offer,
+            countif(event_name = 'BookingConfirmation') as nb_booking_confirmation,
+            countif(event_name = 'HasAddedOfferToFavorites') as nb_add_to_favorites,
+            countif(event_name = 'Share') as nb_share,
```
```suggestion
            countif(event_name = 'Share') as total_share,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            {% if is_incremental() %}
+                -- lag in case session is between two days.
+                event_date
+                between date_sub(date('{{ ds() }}'), interval 3 + 1 day) and date(
```
Pourquoi prendre 1J de plus ? Avec le paramètre d'overwrite on aura bien tous les events de la session à terme dans la table non ?

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_booking.sql:
Cette table est vouée à remplacer "firebase_bookings" ?

lricard-pass replied:
Yes indeed

lricard-pass committed:
Update orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql

Co-authored-by: Jules-Arbelot <115459312+Jules-Arbelot@users.noreply.github.com>

lricard-pass committed:
Update orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql

Co-authored-by: Jules-Arbelot <115459312+Jules-Arbelot@users.noreply.github.com>

lricard-pass committed:
Update orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql

Co-authored-by: Jules-Arbelot <115459312+Jules-Arbelot@users.noreply.github.com>

lricard-pass committed:
Update orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql

Co-authored-by: Jules-Arbelot <115459312+Jules-Arbelot@users.noreply.github.com>

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'Share') as total_share,
+            countif(event_name = 'Screenshot') as nb_screenshot,
+            countif(event_name = 'UserSetLocation') as nb_set_location,
+            countif(event_name = 'ConsultVideo') as nb_consult_video,
```
```suggestion
            countif(event_name = 'ConsultVideo') as total_consult_video,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'Screenshot') as nb_screenshot,
+            countif(event_name = 'UserSetLocation') as nb_set_location,
+            countif(event_name = 'ConsultVideo') as nb_consult_video,
+            countif(event_name = 'ConsultVenue') as nb_consult_venue,
```
```suggestion
            countif(event_name = 'ConsultVenue') as total_consult_venue,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'UserSetLocation') as nb_set_location,
+            countif(event_name = 'ConsultVideo') as nb_consult_video,
+            countif(event_name = 'ConsultVenue') as nb_consult_venue,
+            countif(event_name = 'ConsultVenueMap') as nb_consult_venue_map,
```
```suggestion
            countif(event_name = 'ConsultVenueMap') as total_consult_venue_map,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'ConsultVideo') as nb_consult_video,
+            countif(event_name = 'ConsultVenue') as nb_consult_venue,
+            countif(event_name = 'ConsultVenueMap') as nb_consult_venue_map,
+            countif(event_name = 'ConsultArtist') as nb_consult_artist,
```
```suggestion
            countif(event_name = 'ConsultArtist') as total_consult_artist,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'ConsultVenue') as nb_consult_venue,
+            countif(event_name = 'ConsultVenueMap') as nb_consult_venue_map,
+            countif(event_name = 'ConsultArtist') as nb_consult_artist,
+            countif(event_name = 'screen_view') as nb_screen_view,
```
```suggestion
            countif(event_name = 'screen_view') as total_screen_view,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(event_name = 'screen_view') as nb_screen_view,
+            countif(
+                event_name = 'screen_view' and firebase_screen = 'Home'
+            ) as nb_screen_view_home,
```
```suggestion
            ) as total_screen_view_home,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            ) as nb_screen_view_home,
+            countif(
+                event_name = 'screen_view' and firebase_screen = 'Search'
+            ) as nb_screen_view_search,
```
```suggestion
            ) as total_screen_view_search,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            ) as nb_screen_view_search,
+            countif(
+                event_name = 'screen_view' and firebase_screen = 'Offer'
+            ) as nb_screen_view_offer,
```
```suggestion
            ) as total_screen_view_offer,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            ) as nb_screen_view_offer,
+            countif(
+                event_name = 'screen_view' and firebase_screen = 'Profile'
+            ) as nb_screen_view_profile,
```
```suggestion
            ) as total_screen_view_profile,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            ) as nb_screen_view_profile,
+            countif(
+                event_name = 'screen_view' and firebase_screen = 'Favorites'
+            ) as nb_screen_view_favorites,
```
```suggestion
            ) as total_screen_view_favorites,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(
+                event_name = 'screen_view'
+                and firebase_screen in ('Bookings', 'BookingDetails')
+            ) as nb_screen_view_bookings,
```
```suggestion
            ) as total_screen_view_bookings,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+            countif(
+                firebase_screen = 'SignupConfirmationEmailSent'
+                or event_name = 'ContinueCGU'
+            ) as nb_signup_completed,
```
```suggestion
            ) as total_completed_signup,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+                    'UnderageAccountCreated',
+                    'BeneficiaryAccountCreated'
+                )
+            ) as nb_benef_request_sent,
```
```suggestion
            ) as total_benef_sent_request,
```

Jules-Arbelot commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_visit.sql:
```diff
@@ -0,0 +1,119 @@
+                    'BeneficiaryAccountCreated'
+                )
+            ) as nb_benef_request_sent,
+            countif(event_name = 'login') as nb_login,
```
```suggestion
            countif(event_name = 'login') as total_login,
```

lricard-pass commented on orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_booking.sql:
```diff
@@ -0,0 +1,33 @@
+        **custom_incremental_config(
+            incremental_strategy="insert_overwrite",
+            partition_by={"field": "booking_date", "data_type": "date"},
+            on_schema_change="sync_all_columns",
```
Yes chaud de vos habitudes sur les modèles int @lrainteau-pass @Jules-Arbelot

lricard-pass committed:
Refactor Firebase native visit model column names

- Rename event count columns to use 'total_' prefix for consistency
- Simplify date range calculation in incremental and full load scenarios
- Remove redundant interval day calculation

lricard-pass committed:
Improve formatting and readability of Firebase native visit model

- Apply consistent code formatting with line breaks and indentation
- Enhance readability of complex countif and date range conditions
- Maintain existing logic while improving code structure
